cmd/fleet/handleEnroll.go

```diff
@@ -76,53 +77,44 @@ func (rt Router) handleEnroll(w http.ResponseWriter, r *http.Request, ps httprou
 		return
 	}
 
-	enrollResponse, err := rt.et.handleEnroll(w, r)
+	reqId := r.Header.Get(logger.HeaderRequestID)
 
-	var data []byte
-	if err == nil {
-		data, err = json.Marshal(enrollResponse)
-	}
+	zlog := log.With().
+		Str(EcsHttpRequestId, reqId).
+		Str("mod", kEnrollMod).
+		Logger()
 
-	reqId := r.Header.Get(logger.HeaderRequestID)
+	resp, err := rt.et.handleEnroll(&zlog, w, r)
 
 	if err != nil {
 		cntEnroll.IncError(err)
 		resp := NewErrorResp(err)
 
-		log.WithLevel(resp.Level).
+		zlog.WithLevel(resp.Level).
 			Err(err).
-			Str(EcsHttpRequestId, reqId).
-			Str("mod", kEnrollMod).
 			Int(EcsHttpResponseCode, resp.StatusCode).
 			Int64(EcsEventDuration, time.Since(start).Nanoseconds()).
 			Msg("fail enroll")
 
 		if err := resp.Write(w); err != nil {
-			log.Error().Err(err).Str(EcsHttpRequestId, reqId).Msg("fail writing error response")
+			zlog.Error().Err(err).Msg("fail writing error response")
 		}
 		return
 	}
 
-	var numWritten int
-	if numWritten, err = w.Write(data); err != nil {
-		log.Error().Err(err).Str(EcsHttpRequestId, reqId).Msg("fail send enroll response")
-	}
-
-	cntEnroll.bodyOut.Add(uint64(numWritten))
+	if err = writeResponse(zlog, w, resp, start); err != nil {
+		cntEnroll.IncError(err)
+		zlog.Error().
+			Err(err).
+			Int64(EcsEventDuration, time.Since(start).Nanoseconds()).
+			Msg("fail write response")
 
-	log.Info().
-		Err(err).
-		Str("mod", kEnrollMod).
-		Str("agentId", enrollResponse.Item.ID).
-		Str("policyId", enrollResponse.Item.PolicyId).
-		Str("apiKeyId", enrollResponse.Item.AccessApiKeyId).
-		Str(EcsHttpRequestId, reqId).
-		Int(EcsHttpResponseBodyBytes, numWritten).
-		Int64(EcsEventDuration, time.Since(start).Nanoseconds()).
-		Msg("success enroll")
+		// Remove ghost artifacts; agent will never receive the paylod
+		rt.et.wipeGhosts(r.Context(), zlog, resp)
```

Comment:
the cleanup will stop if the request is cancelled, this will need to be revisited. agreed to address after this PR is merged.